dev/build_release.py

```diff
@@ -211,28 +212,62 @@ def determine_gradle_root(self, name):
       gradle_root = name if name != 'spinnaker' else self.__project_dir
       return gradle_root
 
-  def start_build_target(self, name, target):
-      """Start a subprocess to build the designated target.
+  def start_candidate_release(self, name):
+    """Start a subprocess to build and publish the designated component.
 
-      Args:
-        name [string]: The name of the subsystem repository.
-        target [string]: The gradle build target.
+    This function runs a 'candidate' release using the last git tag as the package
+    version and the Bintray configuration passed through arguments. A 'candidate'
+    release builds the source, packages the debian and jar files, and publishes
+    those to the respective Bintray '$org/$repository'.
 
-      Returns:
-        BackgroundProcess
-      """
-      extra_args = []
-      if name == 'deck' and not 'CHROME_BIN' in os.environ:
-        extra_args.append('-PskipTests')
-
-      # Currently spinnaker is in a separate location
-      gradle_root = self.determine_gradle_root(name)
-      print 'Building {name}...'.format(name=name)
-      return BackgroundProcess.spawn(
-          'Building {name}'.format(name=name),
-          'cd "{gradle_root}"; ./gradlew {target} {extra}'.format(
-              gradle_root=gradle_root, target=target,
-              extra=' '.join(extra_args)))
+    The 'candidate' gradle release task throws a 409 if the package we are trying to
+    publish already exists. We'll publish unique package version using build
+    numbers. These will be transparent to end users since the only meaningful
+    version is the Spinnaker product version.
+
+    We will use -Prelease.useLastTag=true and ensure the last git tag is the
+    version we want to use. This tag has to be of the form 'X.Y.Z-$build' or
+    'vX.Y.Z-$build for gradle to use the tag as the version. This script will
+    assume that the source has been properly tagged to use the latest tag as the
+    package version for each component.
+
+    Args:
+      name [string]: Name of the subsystem repository.
+
+    Returns:
+      BackgroundProcess
+    """
+    jarRepo = self.__options.jar_repo
+    parts = self.__options.bintray_repo.split('/')
```

Comment:
Do these come from gradle properties?
I'm ok with environment variables.
I thought the current gradle plugins used properties.